Add some more detail to the new Headless article (#5417)
GoogleChrome/developer.chrome.com: site/en/articles/new-headless/index.md

Before we dive into the recent Headless improvements, it’s important to understand how the “old” Headless worked. The command-line snippet we showed earlier uses the `--headless` command-line flag, suggesting that Headless is just a mode of operation of the regular Chrome browser. Perhaps surprisingly, this wasn’t actually true. Technically, **the old Headless was [a separate, alternate browser implementation](https://source.chromium.org/chromium/chromium/src/+/main:headless/;drc=c67febd82ae3e18ac8db1397f4ccfa87b0da2ffc)** that happened to be shipped as part of the same Chrome binary. It doesn’t share any of the Chrome browser code in [`//chrome`](https://source.chromium.org/chromium/chromium/src/+/main:chrome/).

> As you might imagine, implementing and maintaining this separate Headless browser came with a lot of engineering overhead — but that wasn’t the only problem. Because Headless was a separate implementation, it had its own bugs and features that weren’t present in headful Chrome. This created a confusing situation where any automated browser test might pass in headful mode but fail in Headless mode, or vice versa — a major pain point for automation engineers. It also precluded any automated testing that relied on having a browser extension installed, for example. The same goes for any other browser-level functionality: unless Headless had its own, separate implementation of it, it was unsupported.

Reviewer: ```suggestion
As you might imagine, implementing and maintaining this separate Headless browser came with a lot of engineering overhead — but that wasn’t the only problem. Because Headless was a separate implementation, it had its own bugs and features that weren’t present in headful Chrome. This created a confusing situation where any automated browser test might pass in headful mode but fail in Headless mode, or vice versa — a major pain point for automation engineers. It also excluded any automated testing that relied on having a browser extension installed, for example. The same goes for any other browser-level functionality: unless Headless had its own, separate implementation of it, it wasn't supported.
```

chrome --headless=new --print-to-pdf https://developer.chrome.com/
```

> Optionally, the `--no-pdf-header-footer` flag can be added to omit the print header (containing the current date and time) and footer (containing the URL and the page number).

Reviewer: ```suggestion
Optionally, you can add the `--no-pdf-header-footer` flag to omit the print header (with the current date and time) and footer (with the URL and the page number).
```

```sh
chrome --headless=new --print-to-pdf --no-pdf-header-footer https://developer.chrome.com/
```

{% Aside %}

> The functionality behind the `--no-pdf-header-footer` flag used to be available via the `--print-to-pdf-no-header` flag instead. Depending on which Chrome version you’re using, you might need to fall back to the old flag name.

Reviewer: ```suggestion
The functionality behind the `--no-pdf-header-footer` flag was previously available via the `--print-to-pdf-no-header` flag. Depending on which Chrome version you’re using, you might need to fall back to the old flag name.
```
